ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,1421 @@
+
+Should there be any breaking changes, it should explicitly describe the migration path. Should there be no possible migration paths, it should instead explain why it is not possible, and why we decided that the benefits are worth breaking compatibility.
+
+After reading this section, a contributor should know the following:
+
+- [ ] Will it be possible to update a Zeebe cluster?
+- [ ] If applicable, what is the update procedure? Is it automated?
+- [ ] Does the ZEP break compatibility in the Go client?
+- [ ] Does the ZEP break compatibility in `zeebe-client`?
+- [ ] Does the ZEP break compatibility in `zeebe-bpmn-model`?
+- [ ] Does the ZEP break compatibility in `zeebe-exporter-api`?
+- [ ] Does the ZEP break compatibility in `zeebe-protocol`?
+- [ ] Does the ZEP break compatibility in `zeebe-gateway-protocol`?
+- [ ] Does the ZEP break compatibility in `zeebe-test`?
+
+-->
+
+The new stream processing changes the records that are written to the stream. It adds new commands and removes some events. 
+
+* new commands for all BPMN elements:
+  * `WorkflowInstance:Activate_Element` 
+  * `WorkflowInstance:Complete_Element`
+  * `WorkflowInstance:Terminate_Element`
+* new events for message boundary/intermediate events:
+  * `WorkflowInstanceSubscription:Opening`
+  * `MessageSubscription:Correlating`
+* new events for call activities:
+  * `WorkflowInstanceCreationIntent:Created`
+* new events for message start events:
+ * `MessageStartEventSubscription:Correlated` (also adds workflow instance key to the record) 
+* removed records:  
+  * `Job:Create`
+  * `Timer:Create`
+  * `Incident:Create`
+  * `WorkflowInstance:Cancel`
+  * `WorkflowInstance:Event_Occurred` if an event (timer, message, error) is triggered for an element
+
+Technically, these changes break the public API of the exporter. However, these records are mainly used for the internal processing, and it is not likely that they're used by a consumer of the stream.    
+
+Beside the records itself, the behavior of the processing changes slightly (i.e. the sequence of records that are written). The processing of a command produces more events and includes more state transitions. Previously, the processing was more fine granular by processing every single event.
+
+TODO: discuss if the changes could be backward-compatible with the current processing (e.g. by starting the processing from a snapshot).
+
+### Impact on Operate and Tasklist
+
+The proposed changes do not make any impact on Operate and Tasklist.
+
+Operate and Tasklist only process records that are considered events in this
+proposal. Only the `MessageStartEventSubscription` records have changed, which
+are not consumed by either Operate and Tasklist. The removed
+`WorkflowInstance:Event_Occurred` is also not consumed by either one. New
+records have no impact, because they weren't consumed yet.
```

Comment:
Well, if they will need to be consumed in the future, this also has an impact on Operate/Task list. Not in the sense of incompatibility, but in the sense of extra work for them.

Reply:
They don't need to consume them to achieve the same behaviour is what I meant. I'll update the text.